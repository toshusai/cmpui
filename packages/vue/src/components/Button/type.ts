export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
