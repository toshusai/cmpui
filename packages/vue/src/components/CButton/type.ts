export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
