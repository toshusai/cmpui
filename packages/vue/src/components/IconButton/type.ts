export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
