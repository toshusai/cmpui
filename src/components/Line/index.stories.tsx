import type { Meta, StoryObj } from "@storybook/react";
import { Line } from ".";

const meta: Meta<typeof Line> = {
  title: "Components/Controls/Line",
  component: Line,
};

export default meta;
type Story = StoryObj<typeof Line>;

export const Basic: Story = {
  render: function Render() {
    return (
      <div
        style={{
          width: 200,
          height: 200,
          position: "relative",
        }}
      >
        <Line
          x1={0}
          y1={0}
          x2={200}
          y2={200}
          strokeWidth={1}
          stroke="var(--cmpui-primary-color)"
        />
      </div>
    );
  },
};
