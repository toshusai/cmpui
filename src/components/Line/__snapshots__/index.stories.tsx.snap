// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Controls/Line Basic smoke-test 1`] = `
<div style="width: 200px; height: 200px; position: relative;">
  <svg width="201"
       height="201"
       viewbox="0 0 201 201"
       style="position: absolute; overflow: visible; pointer-events: none; left: -0.5px; top: -0.5px;"
  >
    <line x1="0.5"
          y1="0.5"
          x2="200.5"
          y2="200.5"
          stroke-width="1"
          stroke="var(--cmpui-primary-color)"
    >
    </line>
  </svg>
</div>
`;
